app/scenes/select_server/select_server.js

```diff
@@ -21,22 +21,46 @@ import logo from 'assets/images/logo.png';
 import Client from 'service/client';
 import RequestStatus from 'service/constants/request_status';
 
-export default class SelectServer extends Component {
+import {isValidUrl, stripTrailingSlashes} from 'app/utils/url';
+
+export default class SelectServer extends PureComponent {
     static propTypes = {
-        serverUrl: React.PropTypes.string.isRequired,
-        server: React.PropTypes.object.isRequired,
-        actions: React.PropTypes.object.isRequired
+        serverUrl: PropTypes.string.isRequired,
+        server: PropTypes.object.isRequired,
+        actions: PropTypes.object.isRequired
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null
+        };
+    }
+
     onClick = () => {
-        Client.setUrl(this.props.serverUrl);
+        const url = this.props.serverUrl;
+        let error = null;
```

Comment:
Or if the else statement is not reached then it will set the error state to null making the error dissapear if it was there in the first place. (I'm thinking clicking for the second time when you fixed the url in the input box)